### PR TITLE
[Enhancement] enhance sql digest

### PR DESCRIPTION
**Pull request by murphyatwork**
## What I'm doing:

1. Parameterization
    1. Literal values
    2. `INSERT WITH LABEL xxx` => `WITH LABEL ?` 
    3. `X IN (1,2,3)`  => ` X IN (?)` 
    4. `VALUES(1,2,3), (4,5,6)` will be parameterized into `VALUES(?,?,?)`

Configuration change:
1. Add a fe configuration `enable_sql_digest` to turn on this functionality, whose default is still false
2. Whether `set enable_sql_digest =true` or set the configuration to true in fe.conf, both can take affect

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -1086,6 +1086,8 @@ public static MaterializedViewRewriteMode parse(String str) {
     @VarAttr(name = CBO_PRUNE_JSON_SUBFIELD_DEPTH, flag = VariableMgr.INVISIBLE)
     private int cboPruneJsonSubfieldDepth = 20;
 
+    // Deprecated by fe.conf/enable_sql_digest
+    @Deprecated
     @VarAttr(name = ENABLE_SQL_DIGEST, flag = VariableMgr.INVISIBLE)
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:1091]: why change it?

Comment (murphyatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java]: fixed

**Review by before-Sunrise — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/common/SqlDigestBuilder.java [hunk 2]
```diff
@@ -14,18 +14,68 @@
...
+                String notStr = (node.isNotIn()) ? "NOT " : "";
+                strBuilder.append(printWithParentheses(node.getChild(0))).append(" ").append(notStr).append("IN ");
+                strBuilder.append("(?)");
+                return strBuilder.toString();
+            }
+        }
+
+        @Override
+        public String visitValues(ValuesRelation node, Void scope) {
+            if (node.isNullValues()) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/common/SqlDigestBuilder.java:55]: why use empty instead of NULL?

Comment (murphyatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/common/SqlDigestBuilder.java:55]: it should be `VALUES(NULL)`

**Review by stephen-shelby — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**